### PR TITLE
[iOS] Login changes for EasyAuth support

### PR DESCRIPTION
**Pull request by phvannor**

### CODE REVIEW

File: sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoLoginTests.m [hunk 1]
```diff
@@ -46,30 +46,36 @@ + (NSArray *)createTests {
     NSMutableArray *result = [[NSMutableArray alloc] init];
```
Comment (pragnagopa) [inline sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoLoginTests.m]: Can you add a test for loginPrefix and alternate login host? Expected result can be 404

**Comment by jwallra** [comment]
What version of WindowsAzureMobileServices.framework did you update to?  I just submitted a PR that also updates it (to 0.3.21).

**Comment by phvannor** [comment]
We don't check in any specific version of the FW, as to what version I tested with, a custom build of the MobileApp branch (which would make it an alpha build of the iOS-3.0.0-beta4 cut)

File: sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoLoginTests.m (hunk 1 above)
Comment (phvannor) [inline sdk/iOS/ZumoE2ETestApp/ZumoE2ETestApp/ZumoLoginTests.m]: I'm going to figure this portion out separately to unblock myself.

**Pull request merged**